app/services/submit_application_choice.rb

```diff
@@ -0,0 +1,23 @@
+class SubmitApplicationChoice
+  def initialize(application_choice)
+    @application_choice = application_choice
+  end
+
+  def call
+    return unless @application_choice.unsubmitted?
```

Comment:
@duncanjbrown  I've written a test as discussed and added this line to prevent re-submitting existing application choices, good catch 🎣

Reply:
This class already handles this condition by raising an error on `submit!`. imo it's the caller's responsibility not to give bad input. Maybe something like `@application_form.application_choices.filter(&:unsubmitted?).each { |choice| SubmitApplicationChoice.new(choice).call }`?

Reply:
Make sense 😄